scripts/irods/test/test_access_time_updates.py

```diff
@@ -59,13 +63,144 @@ def assert_atime_matches_mtime_for_replica(data_object, replica_number):
+            IrodsController().restart(test_mode=True)
+
+            # Show that a new shared memory file exists with the queue name prefix we just set.
+            self.assertTrue(any(f.name.startswith(option_value) for f in shm_directory.iterdir()))
+
+        finally:
+            # Restore the original grid configuration value.
+            self.admin.run_icommand(['iadmin', 'set_grid_configuration', 'access_time', option_name, original_option_value])
+            IrodsController().restart(test_mode=True)
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_grid_configuration_value_for_queue_size_is_honored__issue_8260(self):
+        def get_atime_queue_file_size():
+            shm_directory = Path('/dev/shm')
+            for f in shm_directory.iterdir():
+                if f.name.startswith('irods_access_time_queue_'):
+                    return f.stat().st_size
+            raise ValueError('Could not get size of shared memory file for access time')
+
+        # Capture the original value so it can be restored.
+        option_name = 'queue_size'
+        _, out, _ = self.admin.assert_icommand(['iadmin', 'get_grid_configuration', 'access_time', option_name], 'STDOUT')
+        original_option_value = out.strip()
+
+        try:
+            # Set the queue size to the default value. This requires a server restart
+            # restart to take effect. The grid configuration options for access time
+            # are only loaded on server startup.
+            option_value = '20000'
+            self.admin.assert_icommand(['iadmin', 'set_grid_configuration', 'access_time', option_name, option_value])
+            IrodsController().restart(test_mode=True)
+
+            # Show that a shared memory file exists with the default queue name prefix.
+            old_file_size = get_atime_queue_file_size()
+            self.assertGreater(old_file_size, 0)
+
+            # Now, adjust the queue size so that the shared memory file's size shrinks.
+            option_value = '1000'
+            self.admin.assert_icommand(['iadmin', 'set_grid_configuration', 'access_time', option_name, option_value])
+            IrodsController().restart(test_mode=True)
+
+            # Show that the new queue size has resulted in a smaller file size.
+            self.assertLess(get_atime_queue_file_size(), old_file_size)
+
+        finally:
+            # Restore the original grid configuration value.
+            self.admin.run_icommand(['iadmin', 'set_grid_configuration', 'access_time', option_name, original_option_value])
+            IrodsController().restart(test_mode=True)
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_grid_configuration_value_for_resolution_in_seconds_is_honored__issue_8260(self):
+        def get_atime(data_object):
+            query = f"select DATA_ACCESS_TIME where DATA_NAME = '{data_object}'"
```

Comment:
is this sufficient?   could have multiple objects with this data name, and of course, multiple replicas of any of the matching data objects?

Reply:
okay, might be able to tighten the query a little bit more with a COLL_NAME and DATA_REPL_NUM / DATA_REPL_STATUS, but yes, agreed, targeted in the testing environment seems safe/good enough.

Reply:
Instead of adjusting the query, I changed the file names to match the test names. That guarantees uniqueness regardless of where the tests are run.